packages/flutter/lib/src/material/material_state.dart

```diff
@@ -178,6 +181,94 @@ class _MaterialStateColor extends MaterialStateColor {
   Color resolve(Set<MaterialState> states) => _resolve(states);
 }
 
+/// Defines a [MouseCursor] whose value depends on a set of [MaterialState]s which
+/// represent the interactive state of a component.
+///
+/// This is useful when the widget's intractability varies with its state. For
+/// example, a mouse pointer hovering over a disabled [FlatButton] should not
+/// display [SystemMouseCursors.click], since the button is not clickable. To
+/// solve this, you can use [MaterialStateMouseCursor] to assign a different
+/// cursor (such as [SystemMouseCursors.basic]) when the [FlatButton] is
+/// disabled.
+///
+/// To use a [MaterialStateMouseCursor], you should create a subclass of
+/// [MaterialStateMouseCursor] and implement the abstract `resolve` method.
+///
+/// {@tool snippet}
+///
+/// In this next example, we see how you can create a `MaterialStateMouseCursor` by
+/// extending the abstract class and overriding the `resolve` method.
+///
+/// ```dart
+/// class ButtonCursor extends MaterialStateMouseCursor {
+///   const ButtonCursor();
+///
+///   @override
+///   MouseCursor resolve(Set<MaterialState> states) {
+///     if (states.contains(MaterialState.disabled)) {
+///       return SystemMouseCursors.forbidden;
+///     }
+///     return SystemMouseCursors.click;
```

Comment:
Could be `return super.resolve(states)` per the comment about merging this class with Clickable...